geoh5py/objects/drillhole.py

```diff
@@ -733,60 +721,136 @@ def post_processing(self):
                     self._cells = key_map.reshape((-1, 2)).astype("uint32")
                     self.workspace.update_attribute(self, "cells")
 
+    @staticmethod
+    def validate_survey_values(
+        survey: np.ndarray, end_of_hole: float | None
+    ) -> np.ndarray:
+        """
+        Validate the survey values for desurveying.
 
-def compute_deviation(surveys: np.ndarray) -> np.ndarray:
-    """
-    Compute deviation distances from survey parameters.
-
-    :param surveys: Array of azimuth, dip and depth values.
-    """
-    if surveys is None:
-        raise AttributeError(
-            "Cannot compute deviation coordinates without `survey` attribute."
-        )
-
-    lengths = surveys[1:, 0] - surveys[:-1, 0]
-
-    deviation = []
-    for component in [deviation_x, deviation_y, deviation_z]:
-        dl_in = component(surveys[:-1, 1], surveys[:-1, 2])
-        dl_out = component(surveys[1:, 1], surveys[1:, 2])
-        ddl = np.divide(dl_out - dl_in, lengths, where=lengths != 0)
-        deviation += [dl_in + lengths * ddl / 2.0]
-
-    return np.vstack(deviation).T
-
-
-def deviation_x(azimuth, dip):
-    """
-    Compute the easting deviation.
-
-    :param azimuth: Degree angle clockwise from North
-    :param dip: Degree angle positive down from horizontal
-
-    :return deviation: Change in easting distance.
-    """
-    return np.cos(np.deg2rad(450.0 - azimuth % 360.0)) * np.cos(np.deg2rad(dip))
+        - Repeat first survey point at surface for de-survey interpolation
+        - Repeat last survey point at end of hole for de-survey interpolation
+        - Densify along intervals if greater than 50 m
 
+        :param survey: Array of survey points with columns 'Depth', 'Azimuth', 'Dip'.
+        :param end_of_hole: End of the drillhole in meters.
 
-def deviation_y(azimuth, dip):
-    """
-    Compute the northing deviation.
+        :return: Augmented array of survey points with columns 'Depth', 'Azimuth', 'Dip'.
+        """
+        full_survey = survey.copy()
 
-    :param azimuth: Degree angle clockwise from North
-    :param dip: Degree angle positive down from horizontal
+        # Repeat first survey point at surface for de-survey interpolation
+        if survey[0, 0] != 0.0:
+            full_survey = np.vstack([survey[0, :], full_survey])
+            full_survey[0, 0] = 0.0
+
+        if end_of_hole is not None and end_of_hole > full_survey[-1, 0]:
+            last_row = full_survey[-1, :].copy()
+            last_row[0] = end_of_hole
+            full_survey = np.vstack([full_survey, last_row])
+
+        # Densify along intervals if greater than 50 m
+        deltas = np.diff(full_survey[:, 0])
+        new_points = []
+        for ii, delta in enumerate(deltas):
+            if delta < MAXIMUM_DEPTH_INTERVAL:
+                continue
+
+            depths = np.arange(0, delta, MAXIMUM_DEPTH_INTERVAL)
+            d_azm = (full_survey[ii + 1, 1] - full_survey[ii, 1]) / delta

```

Comment:
ok, if I understand well every points at depth (depth or interval) has its dip, and azimuth, corresponding to the deviation?

Reply:
The "intervals" are independent of data, just based on geometry (collar and survey). Once we have intervals defined, then we can compute xyz from depths at any location.

Reply:
Ok.
We don't have "deviation" along a drillhole in geoh5py right?